core/lib/instrument.c

```diff
@@ -2784,6 +2784,53 @@ dr_get_application_name(void)
 #    endif
 }
 
+dr_error_code_t
+dr_get_error_code(void)
+{
+    dcontext_t *dcontext = get_thread_private_dcontext();
+    return dcontext->client_data->error_code;
+}
+
+int
+dr_num_app_args(void)
+{
+    /* XXX i#2662: Add support for Windows. */
```

Comment:
Should this set the error code to NOT_IMPLEMENTED?

OK I see there's an ASSERT* below, and this function doesn't say it sets the error code.  Should we try to have all new functions set it?  The ASSERT* are debug-build-only, so we could have both.

Reply:
Done.